lib/hawkular/base_client.rb

```diff
@@ -131,18 +131,19 @@ def generate_query_params(params = {})
     # also, this function always remove the slash at the end of the URL, so if your entrypoint is
     # http://localhost/hawkular/inventory/ this function will return http://localhost/hawkular/inventory
     # to the URL
-    # @param entrypoint [String] base path
+    # @param entrypoint [String] base path (URIs are also accepted)
     # @param suffix_path [String] sufix path to be added if it doesn't exist
     # @return [String] URL with path attached to it at the end
     def normalize_entrypoint_url(entrypoint, suffix_path)
+      fail ArgumentError, 'suffix_path must not be empty' if suffix_path.empty?
       strip_path = suffix_path.gsub(%r{/$}, '')
       strip_path.nil? || suffix_path = strip_path
       strip_path = suffix_path.gsub(%r{^/}, '')
       strip_path.nil? || suffix_path = strip_path
-      strip_entrypoint = entrypoint.gsub(%r{/$}, '')
-      strip_path.nil? && strip_entrypoint = entrypoint
+      strip_entrypoint = entrypoint.to_s.gsub(%r{/$}, '')
```

Comment:
I think doing `entrypoint = entrypoint.to_s` above here should eliminate need to do the `to_s` conversions below.